include/boost/geometry/formulas/area_formulas.hpp

```diff
@@ -563,17 +581,22 @@ class area_formulas
     static inline bool crosses_prime_meridian(PointOfSegment const& p1,
                                               PointOfSegment const& p2)
     {
-        CT const pi
-            = geometry::math::pi<CT>();
-        CT const two_pi
-            = geometry::math::two_pi<CT>();
-
-        CT p1_lon = get_as_radian<0>(p1)
-                                - ( floor( get_as_radian<0>(p1) / two_pi )
-                                  * two_pi );
-        CT p2_lon = get_as_radian<0>(p2)
-                                - ( floor( get_as_radian<0>(p2) / two_pi )
-                                  * two_pi );
+        CT const pi = geometry::math::pi<CT>();
+        CT const two_pi = geometry::math::two_pi<CT>();
+
+        CT const lon1r = get_as_radian<0>(p1);
+        CT const lon2r = get_as_radian<0>(p2);
+
+        CT lon12 = lon1r - lon2r;
+        math::normalize_longitude<radian, CT>(lon12);
+
+        if (lon12 == pi || lon12 == -pi)
+        {
+            return true;
+        }
+
+        CT p1_lon = lon1r - ( floor( lon1r / two_pi ) * two_pi );
+        CT p2_lon = lon2r - ( floor( lon2r / two_pi ) * two_pi );
```

Comment:
`math::floor` or `std::floor`?

Reply:
aah that was there already, but you are right, I should be more specific. `std::floor` should be better.